viewer/packages/camera-manager/src/CameraManager.ts

```diff
@@ -0,0 +1,569 @@
+/*!
+ * Copyright 2021 Cognite AS
+ */
+
+import * as THREE from 'three';
+import TWEEN from '@tweenjs/tween.js';
+import { ComboControls } from './ComboControls';
+import { CallbackData, CameraControlsOptions, CameraChangeData, PointerEventDelegate, ControlsState } from './types';
+import { assertNever, EventTrigger, InputHandler } from '@reveal/utilities';
+import { RevealCameraControls } from '..';
+
+export class CameraManager {
+  private readonly _events = {
+    cameraChange: new EventTrigger<CameraChangeData>()
+  };
+
+  private readonly _controls: ComboControls;
+  private readonly _camera: THREE.PerspectiveCamera;
+  private readonly _domElement: HTMLElement;
+  private readonly _inputHandler: InputHandler;
+  private readonly _raycaster: THREE.Raycaster = new THREE.Raycaster();
+
+  private isDisposed = false;
+
+  private readonly _modelRaycastCallback: (x: number, y: number) => Promise<CallbackData>;
+  private _onClick: ((event: MouseEvent) => void) | undefined = undefined;
+  private _onWheel: ((event: MouseEvent) => void) | undefined = undefined;
+
+  private static readonly DefaultAnimationDuration = 600;
+  private static readonly DefaultMinAnimationDuration = 600;
+  private static readonly DefaultMaxAnimationDuration = 2500;
+  private static readonly DefaultMinDistance = 0.1;
+  private static readonly DefaultCameraControlsOptions: Required<CameraControlsOptions> = {
+    mouseWheelAction: 'zoomPastCursor',
+    onClickTargetChange: false
+  };
+
+  private _cameraControlsOptions: Required<CameraControlsOptions> = { ...CameraManager.DefaultCameraControlsOptions };
+
+  public automaticNearFarPlane = true;
+  public automaticControlsSensitivity = false;
+
+  /**
+   * Reusable buffers used by functions in Cognite3dViewer to avoid allocations.
+   */
+  private readonly _updateNearAndFarPlaneBuffers = {
+    cameraPosition: new THREE.Vector3(),
+    cameraDirection: new THREE.Vector3(),
+    nearPlaneCoplanarPoint: new THREE.Vector3(),
+    nearPlane: new THREE.Plane(),
+    corners: new Array<THREE.Vector3>(
+      new THREE.Vector3(),
+      new THREE.Vector3(),
+      new THREE.Vector3(),
+      new THREE.Vector3(),
+      new THREE.Vector3(),
+      new THREE.Vector3(),
+      new THREE.Vector3(),
+      new THREE.Vector3()
+    )
+  };
+
+  constructor(
+    camera: THREE.PerspectiveCamera,
+    domElement: HTMLElement,
+    raycastFunction: (x: number, y: number) => Promise<CallbackData>
+  ) {
+    this._camera = camera;
+    this._domElement = domElement;
+    this._inputHandler = new InputHandler(domElement);
+    this._modelRaycastCallback = raycastFunction;
+    this._controls = new ComboControls(camera, domElement);
+    this._controls.dollyFactor = 0.992;
+    this._controls.minDistance = 0.15;
+    this._controls.maxDistance = 100.0;
```

Comment:
Found out maxDistance is never used inside ComboControls 🙂, removed it and minDistance setting since it is always adjusted in runtime. Default dollyFactor from ComboControls is also feeling alright.